library/Rules/Image.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class Image extends AbstractRule
+{
+    public function validate($input)
+    {
+    	if (is_file($input) || filter_var($input, FILTER_VALIDATE_URL)) {
+     	   return is_array(getimagesize($input));
```

Comment:
It says that this function doesn't require the GD library but not the GD extension, I'm not sure, though, I have to test it. But if you see the paragraph right below this note you will see this:

> This function expects filename to be a valid image file. If a non-image file is supplied, it may be incorrectly detected as an image and the function will return successfully.
> Do not use getimagesize() to check that a given file is a valid image. Use a purpose-built solution such as the Fileinfo extension instead.